cmd/influxd/upgrade/config.go

```diff
@@ -152,7 +152,13 @@ func (cu *configUpgrader) save(config map[string]interface{}, path string) (stri
 		return "", err
 	}
 
-	err := ioutil.WriteFile(path, buf.Bytes(), 0666)
+	// Try writing config to the requested location.
+	var err error
+	err = os.MkdirAll(filepath.Dir(path), 0755)
```

Comment:
possibly simply `err` value handling by limiting its scope.

instead of assigning it before the condition, i might declare it in the `if` block like:


```
if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
  // handle err here.
}
```